plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsRepository.java

```diff
@@ -142,7 +168,14 @@ private HdfsBlobStore createBlobstore(URI blobstoreUri, String path, Settings re
         );
 
         try {
-            return new HdfsBlobStore(fileContext, path, bufferSize, isReadOnly(), haEnabled);
+            return new HdfsBlobStore(
+                fileContext,
+                path,
+                bufferSize,
+                isReadOnly(),
+                haEnabled,
+                replicationFactor != null ? replicationFactor.shortValue() : null
```

Comment:
Thanks for adding the verification logic above, but if someone sets the dfs.replication.max higher than a short value, we'll still have an issue here - maybe we should add a third check above to make sure that the replicationFactor is between 0 and max short as a hard stop, regardless of the configured min and max?

Reply:
Good point! I've added a check that the replication factor is >= 1 and <= max short.

HDFS client lib throws `IllegalArgumentException` with message `Replication must be greater than 0` when it's below 1, but I suppose it's better to also check that ourselves so that we can give a better exception message.